xl-erasure-v1.go

```diff
@@ -479,24 +481,36 @@ func (xl XL) ListDir(volume, dirPath string) (entries []string, err error) {
 	if !isValidVolname(volume) {
 		return nil, errInvalidArgument
 	}
-	// FIXME: need someway to figure out which disk has the latest namespace
-	// so that Listing can be done there. One option is always do Listing from
-	// the "local" disk - if it is down user has to list using another XL server.
-	// This way user knows from which disk he is listing from.
 
-	for _, disk := range xl.storageDisks {
-		if entries, err = disk.ListDir(volume, dirPath); err != nil {
-			continue
-		}
-		for i, entry := range entries {
-			if strings.HasSuffix(entry, slashSeparator) && isLeafDirectoryXL(disk, volume, path.Join(dirPath, entry)) {
-				entries[i] = strings.TrimSuffix(entry, slashSeparator)
+	// Random provides a random number from min to max range.
+	random := func(min, max int) int {
+		rand.Seed(time.Now().Unix())
+		return rand.Intn(max-min) + min
+	}
```

Comment:
If you are seeding it each time based on time, you might as well pick mod over time. Seed is only required to initialize the random# generator. In any case, we dont need a separate function for this. Just call rand.Intn() directly.